src/charm.py

```diff
@@ -284,6 +307,34 @@ def _on_ceph_joined(self, event: RelationJoinedEvent) -> None:  # pragma: no cov
         self._stored.resources_created = True
         self.check_required_relations()
 
+    def _on_ceph_client_joined(self, event: RelationJoinedEvent) -> None:
+        """Use ceph-mon:client relation to request creation of ceph-pools."""
+        if not self.unit.is_leader():
+            # Don't request ceph pool creation from non-leader units
+            logger.info("Skipping Ceph pool creation requests from non-leader unit")
+            self.check_required_relations()
```

Comment:
Shouldn't it be at the beginning of the function? Leader need also check required relations?